debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MysqlDefaultValueIT.java

```diff
@@ -623,4 +629,43 @@ public void timeTypeWithConnectMode() throws InterruptedException {
         Duration duration2 = Duration.between(LocalTime.MIN, LocalTime.from(DateTimeFormatter.ofPattern("HH:mm:ss.SSSSSS").parse("23:00:00.123456")));
         assertThat(schemaI.defaultValue()).isEqualTo(new java.util.Date(io.debezium.time.Time.toMilliOfDay(duration2, MySqlValueConverters::adjustTemporal)));
     }
+
+    @Test
+    @FixFor("DBZ-771")
+    public void columnTypeChangeResetsDefaultValue() throws Exception {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.INITIAL)
+                .build();
+        start(MySqlConnector.class, config);
+
+        // Testing.Print.enable();
+
+        SourceRecords records = consumeRecordsByTopic(EVENT_COUNT);
+
+        SourceRecord record = records.recordsForTopic(DATABASE.topicForTable("DBZ_771_CUSTOMERS")).get(0);
+        validate(record);
+
+        Schema customerTypeSchema = record.valueSchema().fields().get(1).schema().fields().get(1).schema();
+        assertThat(customerTypeSchema.defaultValue()).isEqualTo("b2c");
+
+        // Connect to the DB and issue our insert statement to test.
+        try (MySQLConnection db = MySQLConnection.forTestDatabase(DATABASE.getDatabaseName())) {
+            try (JdbcConnection connection = db.connect()) {
+                // Enable Query log option
+                connection.execute("SET binlog_rows_query_log_events=ON");
+
+                connection.execute("alter table DBZ_771_CUSTOMERS change customer_type customer_type int;");
```

Comment:
Do we have a test for case where both column type AND default value changes?

Reply:
Seems not to be the case. I'll add one.